### PR TITLE
cli(entry): quotes sanitization

### PR DESCRIPTION
**Pull request by dhruvdutt**
**What kind of change does this PR introduce?**
Enhancement/fixes

**Did you add tests for your changes?**
Yes

**Summary**
Add quotes sanitization allowing inputs like `"./src/index"` along with `'./src/index'`

Related #315, #331

### CODE REVIEW

File: lib/init/index.js [hunk 1]
```diff
@@ -7,71 +7,71 @@ const pEachSeries = require("p-each-series");
 
 const runPrettier = require("../utils/run-prettier");
 
-const entryTransform = require("./transformations/entry/entry");
-const outputTransform = require("./transformations/output/output");
+const amdTransform = require("./transformations/other/amd");
```
Comment (ematipico) [inline lib/init/index.js]: Maybe we can consider these eslint rules:
- https://eslint.org/docs/rules/sort-vars
- https://eslint.org/docs/rules/sort-keys

Although it could be a killer eheh

Comment (dhruvdutt) [reply lib/init/index.js]: Sure! I'll do it in a separate PR if we all agree on adding the rule.

File: lib/generators/utils/entry.js [hunk 2]
```diff
@@ -84,7 +84,11 @@ module.exports = (self, answer) => {
 					validate
 				)
 			])
-			.then(singularAnswer => `"${singularAnswer["singularEntry"]}"`);
+			.then(singularEntryAnswer => {
+				let { singularEntry } = singularEntryAnswer;
```
Comment (ev1stensberg) [inline lib/generators/utils/entry.js]: Why is this needed?

Comment (dhruvdutt) [reply lib/generators/utils/entry.js]: You mean destructuring or double quotes support?

Comment (ematipico) [reply lib/generators/utils/entry.js]: Looking at the test, probably to check if entry point name has `\` ?

File: lib/generators/utils/entry.js [hunk 3]
```diff
@@ -59,18 +59,18 @@ module.exports = (self, answer) => {
...
-							propAns[val].indexOf("path") < 0 &&
-							propAns[val].indexOf("process") < 0
+							entryPropAnswer[val].charAt(0) !== "(" &&
+							entryPropAnswer[val].charAt(0) !== "[" &&
+							entryPropAnswer[val].indexOf("function") < 0 &&
+							entryPropAnswer[val].indexOf("path") < 0 &&
+							entryPropAnswer[val].indexOf("process") < 0
 						) {
-							propAns[val] = `"${propAns[val]}.js"`;
+							entryPropAnswer[val] = `\'${entryPropAnswer[val].replace(/"|'/g,"").concat(".js")}\'`;
```
Comment (dhruvdutt) [inline lib/generators/utils/entry.js]: The quote sanitization won't affect `function` as entries as we have a check for it. 

https://github.com/webpack/webpack-cli/blob/8787043beac452f605dc57bfa20aa0c1a545d188/lib/generators/utils/entry.js#L36-L44

**Comment by ematipico** [comment]
@ev1stensberg is this good for you?

**Comment by ev1stensberg** [comment]
Needs rebase, and yes. If it works, I was insecure about the formatting around the quotes. You should test this locally before merging

**Comment by dhruvdutt** [comment]
@ev1stensberg @ematipico Can we please merge this and cover the `.js` part in a different PR? I'll file an issue once this is merged. :)
We can have a dicussion there what extensions we support for paths and whether we should hard bind it to `.js`.

**Comment by ev1stensberg** [comment]
Yes, but fix conflicts

**Comment by dhruvdutt** [comment]
@ev1stensberg Resolved and rebased. PTAL.

File: lib/utils/prop-types.js [hunk 4]
```diff
@@ -6,30 +6,30 @@
...
-	"bail",
-	"cache",
 	"profile",
-	"merge",
-	"parallelism",
 	"recordsInputPath",
 	"recordsOutputPath",
 	"recordsPath",
-	"resolveLoader"
+	"resolve",
```
Comment (ev1stensberg) [inline lib/utils/prop-types.js]: Bad rebase?

Comment (dhruvdutt) [reply lib/utils/prop-types.js]: It includes all properties. I think Git/GitHub diff might not interpret it as sorting but only show changes in code based on line number. That's why we see such confusing diffs. You can verify it from the [current file](https://github.com/webpack/webpack-cli/blob/master/lib/utils/prop-types.js).

**Comment by dhruvdutt** [comment]
@ematipico Can you please do a final review?
